### ISSUE TITLE
Tracking Issue for future-incompatibility lint no_mangle_generic_items

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#158365]
This is the **tracking issue** for the `no_mangle_generic_items` future-compatibility warning and other related errors. The goal of this page is to describe why this change was made and how you can fix code that is affected by it. It also provides a place to ask questions or register a complaint if you feel the change should not be made. For more information on the policy around future-compatibility warnings, see our [breaking change policy guidelines][guidelines].

[guidelines]: https://rustc-dev-guide.rust-lang.org/bug-fix-procedure.html

### What is the warning for?

The warning triggers on a no_mangle attribute on a generic item:
```rust
#[no_mangle]
fn no_mangle_generic<T>() {}
```

This was accidentally accepted, but never made any sense.

### Why was this change made?

The attribute makes no sense and has no effect on generic items.

### Recommendations

Remove the attribute.

### When will this warning become a hard error?

No concrete timeline yet.

### Steps

- [x] Implement the lint: https://github.com/rust-lang/rust/pull/30269
- [ ] Make the lint an FCW
- [ ] Raise lint level to deny
- [ ] Change the lint to report in dependencies
- [ ] Switch to a hard error

**Comment by RalfJung** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/154585
